src/MongoDB/WriteResult.c

```diff
@@ -136,8 +142,9 @@ PHP_METHOD(WriteResult, getUpsertedCount)
 		return;
 	}
 
-
-	RETURN_LONG(intern->write_result.nUpserted);
+	if (bson_iter_init_find(&iter, intern->reply, "nUpserted") && BSON_ITER_HOLDS_INT32(&iter)) {
+		RETURN_LONG(bson_iter_int32(&iter));
+	}
```

Comment:
Nested macros aren't great - some old Windows compilers don't like it, but I don't think we need to care about these any more.